WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -199,7 +203,8 @@ def create
       redirect_to competition_path(@competition)
       return
     end
-    @registration = @competition.registrations.build(registration_params.merge(user_id: current_user.id))
+    @registration = Registration.find_by_user_id_and_competition_id(current_user.id, @competition.id) || @competition.registrations.build(user_id: current_user.id)
```

Comment:
I see. I suppose a more explicit implementation would check if the pre-existing registration is not deleted, and throw an exception. I don't think we actually have to write that code, though.

Bigger concern: does this actually work? This should cause a deleted registration to become undeleted, right? Do we have a test of that? What if a registration was accepted, then deleted, and then the user creates their registration again? We wouldn't want that registration to be automatically accepted, right?